python/lsst/daf/persistence/butlerExceptions.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+class NoMapperException(Exception):
+    pass
+
+class NoResults(RuntimeError):
+    def __init__(self, message, datasetType, dataId):
+        message += ' datasetType:' + datasetType + ' dataId:' + str(dataId)
+        super(NoResults, self).__init__(message)
+
+class MultipleResults(RuntimeError):
+    def __init__(self, message, datasetType, dataId, locations):
+        message += ' datasetType:' + datasetType + ' dataId:' + str(dataId) + ' locations:'
+        for location in locations:
+            message += ' ' + str(location)
+        super(MultipleResults, self).__init__(message)
+        self.locations = locations
+
+class NotImplemented(RuntimeError):
+    pass
```

Comment:
hmm. should inherit from NotImplementedError then? It seems like users should be able to catch only types of Butler exceptions. Is there any pythonic theory of exception inheritance?